src/renderer/services/ethereum/transaction.ts

```diff
@@ -8,22 +9,70 @@ import * as RxOp from 'rxjs/operators'
 
 import { LiveData } from '../../helpers/rx/liveData'
 import * as C from '../clients'
+import { ethRouterABI } from '../const'
 import { ApiError, ErrorId, TxHashLD } from '../wallet/types'
-import { ApproveParams, Client$, IsApprovedLD, TransactionService } from './types'
+import { ApproveParams, CallRouterParams, Client$, TransactionService, IsApprovedLD } from './types'
 
 export const createTransactionService = (client$: Client$): TransactionService => {
   const common = C.createTransactionService(client$)
 
+  const runCallRouterDeposit$ = (client: EthClient, params: CallRouterParams): TxHashLD =>
```

Comment:
Can we have all `deposit` related things in `transaction/deposit` (similar to `transaction/swap`)?

Reply:
Followings are Ethereum specific one so can't make it generic I think.
And I guess, following kind of deposit is only needed for ETH/ERC20? Other chains need similar thing?

Reply:
Yep, you are right. _I've missed that's `services/ethereum/transaction.ts` (but not `services/chain/transaction/xyz`)_